drainer/relay/relayer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package relay
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb-binlog/drainer/translator"
+	"github.com/pingcap/tidb-binlog/pkg/binlogfile"
+	tb "github.com/pingcap/tipb/go-binlog"
+)
+
+var _ Relayer = &relayer{}
+
+// Relayer is the interface for writing relay log.
+type Relayer interface {
+	// WriteBinlog writes binlog to relay log file.
+	WriteBinlog(schema string, table string, tiBinlog *tb.Binlog, pv *tb.PrewriteValue) (tb.Pos, error)
+
+	// GCBinlog removes unused relay log files.
+	GCBinlog(pos tb.Pos)
+
+	// Close releases resources.
+	Close() error
+}
+
+type relayer struct {
+	tableInfoGetter translator.TableInfoGetter
+	binlogger       binlogfile.Binlogger
+	// lastFileSuffix is the file suffix of last written relay log file.
+	lastFileSuffix uint64
+}
+
+// NewRelayer creates a relayer.
+func NewRelayer(dir string, maxFileSize int64, tableInfoGetter translator.TableInfoGetter) (Relayer, error) {
+	// If `dir` is empty, it means relay log is disabled.
+	if len(dir) == 0 {
```

Comment:
return error and check outside if it's disabled, don't call this `NewRelayer`

Reply:
Addressed.